### PR TITLE
FUTURE(ext/net): remove `Deno.(Conn|TlsConn|Listener|TlsListener|UnixConn).prototype.rid`

### PR DESCRIPTION
**Pull request by iuioiua**
Towards #23089

### CODE REVIEW

File: ext/net/01_net.js [hunk 1]
```diff
@@ -100,10 +100,10 @@ class Conn {
...
 
   constructor(rid, remoteAddr, localAddr) {
-    ObjectDefineProperty(this, internalRidSymbol, {
-      enumerable: false,
-      value: rid,
-    });
+    if (internals.future) {
+      ObjectDefineProperty(this, "rid", { value: undefined });
+    }
+    ObjectDefineProperty(this, internalRidSymbol, { value: rid });
```
Comment (iuioiua) [inline ext/net/01_net.js:106]: Minor cleanup: `enumerable` property is `false` by default. Ditto elsewhere.

File: ext/net/01_net.js [hunk 2]
```diff
@@ -100,10 +100,10 @@ class Conn {
   #writable;
 
   constructor(rid, remoteAddr, localAddr) {
-    ObjectDefineProperty(this, internalRidSymbol, {
-      enumerable: false,
-      value: rid,
-    });
+    if (internals.future) {
```
Comment (iuioiua) [inline ext/net/01_net.js:103]: Note: this change also takes care of `TlsConn` and `UnixConn` due to inheritance.

File: ext/net/01_net.js [hunk 3]
```diff
@@ -260,10 +257,10 @@ class Listener {
   #promise = null;
 
   constructor(rid, addr) {
-    ObjectDefineProperty(this, internalRidSymbol, {
-      enumerable: false,
-      value: rid,
-    });
+    if (internals.future) {
```
Comment (iuioiua) [inline ext/net/01_net.js:260]: Note: this also takes care of `TlsListener` due to inheritance.

File: runtime/js/99_main.js [hunk 4]
```diff
@@ -677,6 +677,11 @@ function bootstrapMainRuntime(runtimeOptions, warmup = false) {
       9: future,
     } = runtimeOptions;
 
+    // TODO(iuioiua): remove in Deno v2. This allows us to dynamically delete
+    // class properties within constructors for classes that are not defined
+    // within the Deno namespace.
+    internals.future = future;
```
Comment (bartlomieju) [inline runtime/js/99_main.js:683]: Do the same for workers? :>

File: tests/specs/future/runtime_api/main.js [hunk 5]
```diff
@@ -30,3 +30,40 @@ console.log("Deno.writeAll is", Deno.writeAll);
...
 console.log("Deno.writeSync is", Deno.writeSync);
+
+// TCP
+// Since these tests may run in parallel, ensure this port is unique to this file
+const tcpPort = 4509;
+const tcpListener = Deno.listen({ port: tcpPort });
+console.log("Deno.Listener.prototype.rid is", tcpListener.rid);
+
+const tcpConn = await Deno.connect({ port: tcpPort });
+console.log("Deno.Conn.prototype.rid is", tcpConn.rid);
```
Comment (bartlomieju) [inline tests/specs/future/runtime_api/main.js:41]: Nice tests!

File: runtime/js/99_main.js (hunk 4 above)
Comment (iuioiua) [reply runtime/js/99_main.js:684]: Done

**Review by bartlomieju — APPROVED** [review]
LGTM

**Pull request merged**